MoreLinq.Test/SkipLastWhileTest.cs

```diff
@@ -0,0 +1,112 @@
+#endregion
+
+namespace MoreLinq.Test
+{
+    using System.Collections.Generic;
+    using NUnit.Framework;
+
+    [TestFixture]
+    public class SkipLastWhileTest
+    {
+        [Test]
+        public void IsLazy()
+        {
+            _ = new BreakingSequence<object>().SkipLastWhile(BreakingFunc.Of<object, bool>());
+        }
+
+        [TestCase(SourceKind.Sequence)]
+        [TestCase(SourceKind.BreakingList)]
+        [TestCase(SourceKind.BreakingReadOnlyList)]
+        public void PredicateNeverFalse(SourceKind sourceKind)
+        {
+            using var sequence = TestingSequence.Of(0, 1, 2, 3, 4);
+
+            Assert.That(sequence.ToSourceKind(sourceKind).SkipLastWhile(x => x < 5), Is.Empty);
+        }
+
+        [TestCase(SourceKind.Sequence)]
+        [TestCase(SourceKind.BreakingList)]
+        [TestCase(SourceKind.BreakingReadOnlyList)]
+        public void PredicateNeverTrue(SourceKind sourceKind)
+        {
+            using var sequence = TestingSequence.Of(0, 1, 2, 3, 4);
+
+            sequence.ToSourceKind(sourceKind)
+                    .SkipLastWhile(x => x == 100)
+                    .AssertSequenceEqual(0, 1, 2, 3, 4);
+        }
+
+        [TestCase(SourceKind.Sequence)]
+        [TestCase(SourceKind.BreakingList)]
+        [TestCase(SourceKind.BreakingReadOnlyList)]
+        public void PredicateBecomesTruePartWay(SourceKind sourceKind)
+        {
+            using var sequence = TestingSequence.Of(0, 1, 2, 3, 4);
+
+            sequence.ToSourceKind(sourceKind)
+                    .SkipLastWhile(x => x > 2)
+                    .AssertSequenceEqual(0, 1, 2);
+        }
+
+        [TestCase(SourceKind.Sequence)]
+        // [TestCase(SourceKind.BreakingList)]
+        // [TestCase(SourceKind.BreakingReadOnlyList)]
+        public void NeverEvaluatesPredicateWhenSourceIsEmpty(SourceKind sourceKind)
+        {
+            using var sequence = TestingSequence.Of<int>();
+
+            Assert.That(sequence
+                .ToSourceKind(sourceKind)
+                .SkipLastWhile(BreakingFunc.Of<int, bool>()), Is.Empty);
+        }
+
+        [TestCase(SourceKind.Sequence)]
+        [TestCase(SourceKind.BreakingList)]
+        [TestCase(SourceKind.BreakingReadOnlyList)]
+        public void UsesCollectionCountAtIterationTime(SourceKind sourceKind)
+        {
+            var list = new List<int> { 1, 2, 3, 4 };
+            var result = list.ToSourceKind(sourceKind).SkipLastWhile(x => x > 2);
+            list.Add(5);
+            result.AssertSequenceEqual(1, 2);
+        }
+
+        [TestCase(SourceKind.Sequence)]
+        [TestCase(SourceKind.BreakingList)]
+        [TestCase(SourceKind.BreakingReadOnlyList)]
+        public void OptimizedForCollections(SourceKind sourceKind)
+        {
+            var sequence = new[] { 1, 2, 3, 4, 5, 6, 7, 8, 9, 10 }.ToSourceKind(sourceKind);
+
+            sequence.SkipLastWhile(x => x > 7)
+                    .AssertSequenceEqual(1, 2, 3, 4, 5, 6, 7);
+        }
+
```

Comment:
Unless I've missed something, this seems redundant with `PredicateBecomesTruePartWay` and therefore can be completely removed:

```suggestion
```

There is nothing apparent here about the optimisation.